swift/apple/FirezoneKit/Sources/FirezoneKit/Managers/VPNConfigurationManager.swift

```diff
@@ -93,116 +52,113 @@ public class VPNConfigurationManager {
     manager.localizedDescription = bundleDescription
     manager.protocolConfiguration = protocolConfiguration
 
-    // Save the new VPN configuration to System Preferences and reload it,
-    // which should update our status from nil -> disconnected.
-    // If the user denied the operation, the status will be .invalid
-    do {
-      try await manager.saveToPreferences()
-      try await manager.loadFromPreferences()
-      self.manager = manager
-    } catch let error as NSError {
-      if error.domain == "NEVPNErrorDomain" && error.code == 5 {
-        // Silence error when the user doesn't click "Allow" on the VPN
-        // permission dialog
-        Log.info("VPN permission was denied by the user")
-
-        return
-      }
+    try await manager.saveToPreferences()
+    try await manager.loadFromPreferences()
 
-      throw error
-    }
+    return manager
   }
 
-  func loadFromPreferences(
-    vpnStateUpdateHandler: @escaping @MainActor (NEVPNStatus, Settings?, String?, NEProviderStopReason?) async -> Void
-  ) async throws {
+  static func loadFromPreferences() async throws -> NETunnelProviderManager? {
```

Comment:
Same here, this should just return `VPNConfigurationManager`

Reply:
👍